parser/prism/Translator.cc

```diff
@@ -737,9 +741,33 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node, bool preserveCon
                         supportedBlock = true;
                     }
                 } else {
-                    // `PM_BLOCK_ARGUMENT_NODE` is not supported yet.
+                    ENFORCE(PM_NODE_TYPE_P(prismBlock, PM_BLOCK_ARGUMENT_NODE)); // the `&b` in `a.map(&b)`
+
+                    auto *bp = down_cast<pm_block_argument_node>(prismBlock);
+
                     blockPassNode = translate(prismBlock);
-                    supportedBlock = false;
+
+                    if (bp->expression) {
```

Comment:
This area is analogous to this logic:

https://github.com/sorbet/sorbet/blob/700540c446e5a5fa4e0cb2ba42ddf61a03f3317b/ast/desugar/Desugar.cc#L856-L881